### PR TITLE
Load app definition files from Golem releases CDN

### LINKED ISSUE #5099: Load remote app_definition files
As a golem factory we want to add app_definition.json files to the releases CDN so we do not need a golem release to add a new app

- Every 24h ( time is changeable ) check the release CDN and download new app definitions.
- Send out event when new app definition is found, encouraging the user to enable it.

### PR DESCRIPTION
**Pull request by kmazurek**
Resolves: https://github.com/golemfactory/golem/issues/5099

Adds logic for downloading app definition JSON files from CDN in Task API `AppManager`. The definition files are distinguished based on their file name (which, among other things, should contain the hash of the file's contents). Updating apps is done upon `AppManager` init and then periodically (once a day using `DailyJobsService`). For every new definition downloaded an RPC event is fired (`evt.apps.new_definition`).

### CODE REVIEW

**Review by maaktweluit — APPROVED** [review]
LGTM!

File: golem/core/variables.py [hunk 1]
```diff
@@ -75,6 +75,8 @@
 # How long should peer be banned when failing on resources (seconds)
 ACL_BLOCK_TIMEOUT_RESOURCE = 2 * 3600  # s
 
+APP_DEFINITIONS_CDN_URL = 'https://golem-app-definitions.s3.eu-central-1.amazonaws.com/'    # noqa pylint: disable=line-too-long
```
Comment (shadeofblue) [inline golem/core/variables.py]: shouldn't that rather be some subdomain in `golem.network` that would redirect to s3? ...

Comment (kmazurek) [reply golem/core/variables.py]: @etam ^

Comment (shadeofblue) [reply golem/core/variables.py]: or maybe @ktomala ?

**Pull request merged**